### DISCUSSION TITLE
404 route not found on "as-needed" strategy

### DISCUSSION THREAD

**Discussion by mehrdad-shokri** [amannn/next-intl#1210] [General]
I'm switching from the default routing strategy to "as-needed" strategy. I have a ``[locale]`` directory in the root that catches the route segment. after switching to "as-needed" I get 404 when trying to visit ``/dashboard/chats`` I guess that is expected because Next.js can't find that route (it should start with a route segment) I searched though documentation and it's not mentioned how to solve this issue. Can someone elaborate on this? Thank you

**Comment by amannn** [comment]
My first guess would be that this troubleshooting section could help you: [Unable to find next-intl locale because the middleware didn't run on this request.](https://next-intl-docs.vercel.app/docs/routing/middleware#unable-to-find-locale)
